### ISSUE TITLE
`set_code_hash` must collect immutables

### ISSUE THREAD

**Issue by athei** [paritytech/polkadot-sdk#6717]
With the introduction of immutable variables we can no longer just change the code hash of an existing contract. The reason is that immutable variables are associated with the contract and not with a code hash.

Currently, this is why `set_code_hash` is disabled. In order to fix this we need to run the constructor of the new code hash and overwrite the immutables of the calling current contract. Then we can enable the function again.

Not needed for the minimal launch because this function doesn't exist in Solidity and hence can only be made available via pre-compiles anyways.

**Comment by Zebedeusz** [comment]
@athei I'm up for helping with this issue or any of the smart contract's ones. Is this one a good fit?

**Comment by athei** [comment]
Yes. Please let me know if you need more guidance and feel free to assign yourself when you are about to start.

**Comment by 0xRVE** [comment]
closing this because `set_code_hash` was removed in this PR: https://github.com/paritytech/polkadot-sdk/pull/10517
